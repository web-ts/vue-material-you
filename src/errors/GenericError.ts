export default class GenericError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "[VueMaterialYou] Error";
  }
}
