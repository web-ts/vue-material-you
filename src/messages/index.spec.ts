import GenericError from "@/errors/GenericError";
import { describe, expect, it } from "vitest";
import { getMessage, messages, setMessages } from ".";

describe("messages", () => {
  it("should have defaults", () => {
    expect(messages.value).toMatchSnapshot();
  });

  it("should merge messages", () => {
    setMessages({ dialog: { testMessage: "Test Message" } });
    expect(messages.value).toMatchSnapshot();
  });

  it("should get a message", () => {
    expect(getMessage("dialog.testMessage").value).toEqual("Test Message");
  });

  it("should throw an error if key is not found", () => {
    let error: unknown;

    try {
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      const message = getMessage("dialog.nonExistent").value;
    } catch (e) {
      error = e;
    }
    expect(error).toBeInstanceOf(GenericError);
  });
});
