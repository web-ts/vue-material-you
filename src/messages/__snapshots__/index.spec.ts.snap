// Vitest Snapshot v1

exports[`messages > should have defaults 1`] = `
{
  "dialog": {
    "actions": {
      "cancel": "Cancel",
    },
  },
}
`;

exports[`messages > should merge messages 1`] = `
{
  "dialog": {
    "actions": {
      "cancel": "Cancel",
    },
    "testMessage": "Test Message",
  },
}
`;
